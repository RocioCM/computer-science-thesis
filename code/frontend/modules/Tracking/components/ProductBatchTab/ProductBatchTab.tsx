import React from 'react';
import { ProductBottlesBatch } from '../../types';

interface Props {
  batch?: ProductBottlesBatch | null;
  options: { label: string; value: number }[];
  handleOption: (optionId: number) => void;
}

const ProductBatchTab: React.FC<Props> = ({ batch, options, handleOption }) => {
  return batch ? (
    <div className="h-full w-full overflow-auto hide-scroll relative">
      {/* Información General */}
      <div>
        <div className=" space-y-2">
          <h3 className="text-lg font-semibold border-b pb-2">
            Información General
          </h3>
          <p>
            <span className="font-semibold">Código de seguimiento:</span>{' '}
            {batch.trackingCode || 'Sin código'}
          </p>
          <p>
            <span className="font-semibold">Cantidad de envases:</span>{' '}
            {batch.quantity}
          </p>
          <p>
            <span className="font-semibold">Vendidas:</span>{' '}
            {batch.quantity - batch.availableQuantity}/{batch.quantity}
          </p>
          <p>
            <span className="font-semibold">Fecha de recepción:</span>{' '}
            {new Date(batch.createdAt).toLocaleDateString()}
          </p>
        </div>
      </div>
    </div>
  ) : options ? (
    <div className="h-full w-full overflow-auto hide-scroll relative pt-2">
      <h3 className="text-lg font-semibold border-b pb-2">
        Productos derivados:
      </h3>
      <ul className="mt-2 space-y-2 list-disc list-inside">
        {options.map((option) => (
          <li
            key={option.value}
            onClick={() => handleOption(option.value)}
            className="
					cursor-pointer underline"
          >
            <span className="font-semibold">{option.label}</span>
          </li>
        ))}
      </ul>
    </div>
  ) : null;
};

export default ProductBatchTab;
