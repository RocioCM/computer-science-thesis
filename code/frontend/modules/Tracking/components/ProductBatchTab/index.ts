import ProductBatchTab from './ProductBatchTab';

export default ProductBatchTab;
