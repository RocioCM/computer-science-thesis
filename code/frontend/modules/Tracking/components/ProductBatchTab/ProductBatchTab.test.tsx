import { render } from '@testing-library/react';
import ProductBatchTab from './ProductBatchTab';

describe('Tracking - ProductBatchTab', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches snapshot with batch', () => {
    const { container } = render(
      <ProductBatchTab
        batch={{
          id: 1,
          owner: '0x1234567890abcdef1234567890abcdef12345678',
          quantity: 100,
          availableQuantity: 80,
          originBaseBatchId: 2,
          trackingCode: 'TRACK123',
          createdAt: new Date('2025-04-13T13:00:00').toISOString(),
        }}
        options={[]}
        handleOption={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('Matches snapshot with options', () => {
    const { container } = render(
      <ProductBatchTab
        batch={null}
        options={[
          {
            label: 'Option 1',
            value: 1,
          },
          {
            label: 'Option 2',
            value: 2,
          },
        ]}
        handleOption={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
