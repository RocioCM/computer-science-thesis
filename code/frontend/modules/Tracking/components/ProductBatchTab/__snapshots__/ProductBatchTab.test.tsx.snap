// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking - ProductBatchTab Matches snapshot with batch 1`] = `
<div>
  <div
    class="h-full w-full overflow-auto hide-scroll relative"
  >
    <div>
      <div
        class=" space-y-2"
      >
        <h3
          class="text-lg font-semibold border-b pb-2"
        >
          Información General
        </h3>
        <p>
          <span
            class="font-semibold"
          >
            Código de seguimiento:
          </span>
           
          TRACK123
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Cantidad de envases:
          </span>
           
          100
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Vendidas:
          </span>
           
          20
          /
          100
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Fecha de recepción:
          </span>
           
          13/4/2025
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tracking - ProductBatchTab Matches snapshot with options 1`] = `
<div>
  <div
    class="h-full w-full overflow-auto hide-scroll relative pt-2"
  >
    <h3
      class="text-lg font-semibold border-b pb-2"
    >
      Productos derivados:
    </h3>
    <ul
      class="mt-2 space-y-2 list-disc list-inside"
    >
      <li
        class="        cursor-pointer underline"
      >
        <span
          class="font-semibold"
        >
          Option 1
        </span>
      </li>
      <li
        class="        cursor-pointer underline"
      >
        <span
          class="font-semibold"
        >
          Option 2
        </span>
      </li>
    </ul>
  </div>
</div>
`;
